module1/tasks.js

```diff
@@ -0,0 +1,189 @@
+// ==== Module#1 Basics ====
+mocha.setup('bdd');
+const expect = chai.expect;
+
+// ==== Change the capitalization of all letters in a string ====
+const changeCase = (str) => { 
+  let result = '';
+  for (let i = 0; i < str.length; i++) {
+    const char = str.charAt(i);
+    if(isUpperCase(char)) {
+      result += char.toLowerCase();
+    } else {
+      result += char.toUpperCase();
+    }
+  }
+  return result;
+};
+
+function isUpperCase(char) {
+  return char == char.toUpperCase();
+}
+function convertChar(c) {
+  return isUpperCase(c) ? c.toLowerCase() : c.toUpperCase();
+}
+
+const changeCase2 = (str) => {
+  return [...str]  //transform string to array of chars
+    .map(c => convertChar(c))
+    .join("");  //collect
+};
+
+
+describe('changeCase', () => {
+  it('should change the capitalization of all letters in a given string', () => {
+    expect(changeCase('21century')).to.equal('21CENTURY');
+    expect(changeCase('Hybris')).to.equal('hYBRIS');
+    expect(changeCase2('21century')).to.equal('21CENTURY');
+    expect(changeCase2('Hybris')).to.equal('hYBRIS');
+  });  
+});
+
+// ==== Filter out the non-unique values in an array ====
+const filterNonUnique = (arr) => {
+  let map = new Map();
+  for(let elem of arr) {
+    if(map.has(elem)) {
+      map.set(elem, true);
+    } else {
+      map.set(elem, false);
+    }
+  }
+  return [...map]
+    .filter( ([key, value]) => value == false )
+    .map(([key, value]) => key);
+};
+
+describe('filterNonUnique', () => {
+  it('should filter out non-unique values in an array', () => {
+    expect(filterNonUnique([1, 2, 2, 3, 4, 4, 5])).to.eql([1,3,5]);
+    expect(filterNonUnique([1, 2, 3, 4])).to.eql([1,2,3,4]);
+  });
+});
+
+// ==== Sort string in alphabetical order ====
+const alphabetSort = (str) => {return[...str].sort().join("")};
+
+describe('alphabetSort', () => {
+  it('should accept a string type only', () => {
+    expect(() => alphabetSort()).to.throw();
+    expect(() => alphabetSort('Text')).not.to.throw();
+  });
+
+  it('should convert the letters of a given string in alphabetical order', () => {
+    expect(alphabetSort('Python')).to.equal('Phnoty');
+  });
+});
+
+// ==== Get min integer ====
+const getSecondMinimum = (arr) => {return arr.sort()[1]};
+
+describe('getSecondMinimum', () => {
+  it('should get array of integers and return second minimum value', () => {
```

Comment:
Make this work please

`expect(getSecondMinimum([])).to.equal(undefined);`
`expect(getSecondMinimum([10])).to.equal(undefined);`